test/rules/no-unnecessary-else/test.ts.lint

```diff
@@ -0,0 +1,150 @@
+const testReturn = () => {
+    if () {
+       return;
+       ~~~~~~~  [return]
+    } else {
+       return;
+    }
+}
+
+const testReturn = () => {
+    if () {
+        if () {
+        return;
+        ~~~~~~~  [return]
+        } else {
+            return;
+        }
+    }
+}
+
+//valid
+const testReturn = () => {
+    if () {
+       return;
+    }
+    return;
+}
+
+const testReturn = () => {
+    if () {
+       return;
+       ~~~~~~~  [return]
+    } else if {
+        if () {
+            return ;
+        } else {
+            return ;
+        }
+    }
+    return;
+}
+
+const testReturn = () => {
+    if () {
+       return;
+    }
+    if () {
+        return ;
+        ~~~~~~~~  [return]
+    } else {
+        return ;
+    }
+}
+
+//valid
+const testReturn = () => {
+    if () {
+       return;
+    }
+    if () {
+        return ;
+    }
+    return ;
+}
+
+function testThrow () {
+   if () {
+       throw "error";
+       ~~~~~~~~~~~~~~  [throw]
+   } else {
+   }
+}
+
+const testThrow = () => {
+    switch () {
+        case :
+            break;
+        case :
+            if () {
+                throw "error";
+                ~~~~~~~~~~~~~~  [throw]
+            } else {
+                break;
+            }
+        default :
+            break;
+    }
+}
+
+//valid
+function testThrow () {
+   if () {
+       throw "error";
+   }
+}
+
+function testContinue () {
+    for ( ) { 
+        if () {
+            continue ;
+            ~~~~~~~~~~  [continue]
+        } else {
+        }
+    }
+}
+
+//valid
+function testContinue () {
+    for ( ) { 
+        if () {
+            continue ;
+        }
+    }
+}
+
+function testBreak () {
+   if () {
+       break ;
+       ~~~~~~~  [break]
+   } else {
+   }
+}
+
+const testBreak = () => {
+    switch () {
```

Comment:
It's invalid to have empty contents in `case`, `if`, and `switch` statements. Rules can behave unpredictably when there are syntax errors, so tests should have valid syntax whenever possible.

Reply:
Done https://github.com/palantir/tslint/pull/4502/commits/89c814eb7648cbe716af317adb05add694305430